File: MultiLanguageSandbox/src/test/deps/perl/49.pl

sub ascii_code {
    my ($char) = @_;
    return ord($char);
}


sub check {
    my $func = shift;
    die "Test failed!" unless &$func('A') == 65;
    die "Test failed!" unless &$func('B') == 66;
    die "Test failed!" unless &$func('C') == 67;
    die "Test failed!" unless &$func('a') == 97;
    die "Test failed!" unless &$func('b') == 98;
    die "Test failed!" unless &$func('c') == 99;
    die "Test failed!" unless &$func('0') == 48;
    die "Test failed!" unless &$func('1') == 49;
    die "Test failed!" unless &$func('2') == 50;
}

check(\&ascii_code);

File: MultiLanguageSandbox/src/test/deps/haskell/49.hs
charToAscii :: Char -> Int
charToAscii c = fromEnum c
check :: (Char -> Int) -> IO ()
check charToAsciiFunc = do
  let testCases = 
        [ ('A', 65),
          ('a', 97),
          ('0', 48),
          ('$', 36),
          ('9', 57)
        ]

  mapM_ (\(char, expected) -> testCase char expected) testCases
  where
    testCase char expected = do
      let result = charToAsciiFunc char
      putStrLn $
        "Input: " ++ show char ++
        ", Expected: " ++ show expected ++ ", Result: " ++ show result ++
        if result == expected then " (Pass)" else error "Test failed. Exiting..."

main :: IO ()
main = check charToAscii